typespec-tests/src/main/java/com/cadl/enumservice/models/Operation.java

```diff
@@ -202,4 +192,56 @@ public ColorModel getColorModelValue() {
     public Unit getUnitValue() {
         return this.unitValue;
     }
+
+    @Override
+    public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
+        jsonWriter.writeStartObject();
+        jsonWriter.writeStringField("name", this.name == null ? null : this.name.toString());
+        jsonWriter.writeBooleanField("best", this.best);
+        jsonWriter.writeLongField("age", this.age);
+        jsonWriter.writeLongField("priority", this.priority == null ? null : this.priority.toLong());
+        jsonWriter.writeStringField("color", this.color == null ? null : this.color.toString());
+        jsonWriter.writeDoubleField("unit", this.unit == null ? null : this.unit.toDouble());
+        jsonWriter.writeLongField("priorityValue", this.priorityValue == null ? null : this.priorityValue.toLong());
```

Comment:
```ts
model Operation {
  name: "Read" | "Write";
  best: true;
  age: 50;
  priority: Priority;
  color: ColorModel;
  unit: Unit;
  priorityValue: Priority.Low;
  colorValue: Color.Green;
  colorModelValue: ColorModel.Blue;
  unitValue: Unit.Milligram;
}
```
I wonder if we need the `null` check here. I think for this model, `priorityValue` should only be `Priority.Low`, cannot be `null`. If the field in tsp is `priorityValue: Priority.Low | null`, then it can be `null`.

Reply:
The `null` check isn't needed but I'd rather get this changed in a future PR as this doesn't do anything and will likely be handled by JIT.